pallets/score-staking/src/lib.rs

```diff
@@ -116,6 +119,8 @@ pub mod pallet {
 		type AccountIdConvert: AccountIdConvert<Self>;
 		// For extrinsics that should only be called by origins from TEE
 		type TEECallOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+		#[pallet::constant]
+		type MaxIDGraphAccountsPerCall: Get<u16>;
```

Comment:
Sorry my bad.

We have `MaxScoreUserCount` already and we do iterate over them all in the original `on_initialize` hook. Based on that it should be no problem if we do the whole loop in an extrinsic, so we don't need this constant and batched call in the worker.

But we do need to specify the (max) weights in the extrinsic weight macro - this is the important

Reply:
If we call `distribute` in batch and again we might fall into the problem of inconsistent calculation context

Reply:
I'd still propose a different paradigm, instead of calling `distribute` ext that accepts a vector with potentially 1M entries (that's the `MaxScoreUserCount` runtime binding), can we:
- update the staking amount in batch, with a batch size 500 (1000 might be a bit aggressive, 750 is the limit that we found out last time with batch balance.transfer)
- when it's done, call `distribute`
- within `distribute`, we iterate **all** entrieis in `Scores` and do the reward calculation/distribution in this atomic extrinsic

How does this sound to you?

Reply:
No worries, that's actually what I had in mind with the initial implementation (when I added a new field to the `ScorePayment`), but back then It was not clear to me how the logic was supposed to be. I'll refactor it to do that